src/main/java/core/basesyntax/suppliers/FigureSupplier.java

```diff
@@ -0,0 +1,50 @@
+package core.basesyntax.suppliers;
+
+import core.basesyntax.enums.Color;
+import core.basesyntax.enums.Shape;
+import core.basesyntax.figures.Circle;
+import core.basesyntax.figures.Figure;
+import core.basesyntax.figures.IsoscelesTrapezoid;
+import core.basesyntax.figures.Rectangle;
+import core.basesyntax.figures.RightTriangle;
+import core.basesyntax.figures.Square;
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int BOUND = 10;
+    private static final int START = 1;
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getDefaultFigure() {
+        return new Circle(Color.WHITE, 10);
+    }
+
+    public Figure getRandomFigure() {
+        int index = new Random().nextInt(Shape.values().length);
+        Shape figure = Shape.values()[index];
+        Color color = colorSupplier.getRandomColor();
+
+        switch (figure) {
+            case Square:
+                double side = random.nextInt(BOUND) + START;
+                return new Square(color, side);
```

Comment:
```suggestion
            case Square:
                return getRandomSquare();
```
lets add private methods to get each Random figure - this would make your code easier to read